client/asset/btc/spv.go

```diff
@@ -889,9 +917,12 @@ func (w *spvWallet) connect(ctx context.Context, wg *sync.WaitGroup) error {
 	// Nanny for the spendingTxs cache. We'll keep the cache entries for 2 hours
 	// past their last access time.
 	wg.Add(1)
-	teardownWaitGroup.Add(1)
 	go func() {
-		defer teardownWaitGroup.Done()
+		if err := logNeutrino(w.netDir); err != nil {
+			w.log.Warnf("error initializing btcwallet+neutrino logging: %v", err)
+		} else {
+			defer unlogNeutrino()
```

Comment:
At the top of the defer stack, `wg.Done` is going to happen before this.  Think we should put `defer wg.Done` as the first line in this closure.